internal/services/web/app_service_certificate_resource.go

```diff
@@ -238,13 +240,22 @@ func resourceAppServiceCertificateSchema() map[string]*pluginsdk.Schema {
 			ValidateFunc: validation.NoZeroValues,
 		},
 
+		"key_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateKeyVaultID,
+			RequiredWith: []string{"key_vault_secret_id"},
```

Comment:
If we set this then `key_vault_id` will always need to be set even if users are using a secret from a key vault in the same subscription, this could also constitute a breaking change so this should probably be removed.
```suggestion
```

Reply:
The `RequiredWith: []string{"key_vault_secret_id"},` is to limit the `key_vault_id` can only be set if `key_vault_secret_id` has been set. Otherwise it has no meaning if only `key_vault_id` is set. 

By test `TestAccAppServiceCertificate_KeyVault`, if `key_vault_id` is not specified, there won't be validation error.